### ISSUE TITLE
Task management system

### ISSUE THREAD

**Issue by TheSecondReal0** [opensuspect/opensuspect-legacy#103]
A system to manage what needs to be done before a task is completed and what happens after it is completed. Features to include:
- Keep track of which tasks are accessible to the player (based on prerequisite tasks or player location)
- Prerequisites (allow for tasks require certain conditions to be met before they can be completed. For instance, a task to open a door which requires you to complete another task where you grab a key)
- Effects (opening doors, unlocking another task, etc.. Will need knowledge of the current map for dynamic map objects like doors.)
